### ISSUE TITLE
external_ref/id not handled correctly

### ISSUE THREAD

**Issue by nr23730** [ehrbase/ehrbase#217]
When I execute the following AQL query `SELECT e/ehr_status/subject/external_ref FROM EHR e` I get the following result:
```
{
  "q": "SELECT e/ehr_status/subject/external_ref FROM EHR e",
  "columns": [
    {
      "path": "/ehr_status/subject/external_ref",
      "name": "#0"
    }
  ],
  "rows": [
    [
      {
        "id": {
          "_type": "GENERIC_ID",
          "value": "42",
          "scheme": "DEMOGRAPHIC"
        },
        "type": "PARTY",
        "_type": "PARTY_REF",
        "namespace": "test"
      }
    ]
  ]
}
```
However querying `SELECT e/ehr_status/subject/external_ref/id FROM EHR e` gives the exact same result rows. This causes querying all subsets like `SELECT e/ehr_status/subject/external_ref/id/value FROM EHR e` to be null.

**Comment by ppazos** [comment]
Please check the external_ref is not null in the data you are querying, since by the RM it is not mandatory https://specifications.openehr.org/releases/RM/latest/common.html#_overview_3

**Comment by nr23730** [comment]
> 
> 
> Please check the external_ref is not null in the data you are querying, since by the RM it is not mandatory https://specifications.openehr.org/releases/RM/latest/common.html#_overview_3

Thank you for your reply! As you see in the given example, the value is obviously `42` so it's definitely not null. ;)

**Comment by ppazos** [comment]
yes, but what is the complete EHR_STATUS? can you query that and paste the complete result? thanks

**Comment by nr23730** [comment]
```
{
  "q": "SELECT e/ehr_status FROM EHR e",
  "columns": [
    {
      "path": "/ehr_status",
      "name": "#0"
    }
  ],
  "rows": [
    [
      {
        "name": {
          "value": "EHR Status"
        },
        "_type": "EHR_STATUS",
        "subject": {
          "_type": "PARTY_IDENTIFIED",
          "external_ref": {
            "id": {
              "_type": "GENERIC_ID",
              "value": "123",
              "scheme": "DEMOGRAPHIC"
            },
            "_type": "PARTY_REF",
            "namespace": "test"
          }
        },
        "is_queryable": true,
        "is_modifiable": true,
        "archetype_node_id": "openEHR-EHR-EHR_STATUS.generic.v1"
      }
    ]
  ]
}
```

**Comment by ppazos** [comment]
thanks, it seems you have just one EHR, is that correct?

also, is GET /ehr/uid giving you a consistent result as that one from the AQL? (I mean the same data in the status)

**Comment by nr23730** [comment]
Yes, just a single EHR. I started off a fresh database a few days ago. 

GET/ehr/ehr_id gives me the following result:
```
{
  "compositions": null,
  "contributions": null,
  "system_id": {
    "_type": "HIER_OBJECT_ID",
    "value": "10f4b306-9431-465f-aaff-de5baabd7aba"
  },
  "ehr_id": {
    "_type": "HIER_OBJECT_ID",
    "value": "3fe034c6-635a-4a6a-b690-94cd219f0359"
  },
  "ehr_status": {
    "_type": "EHR_STATUS",
    "name": {
      "_type": "DV_TEXT",
      "value": "EHR Status"
    },
    "subject": {
      "_type": "PARTY_SELF",
      "external_ref": {
        "_type": "PARTY_REF",
        "namespace": "test",
        "id": {
          "_type": "HIER_OBJECT_ID",
          "value": "42"
        }
      }
    },
    "archetype_node_id": "openEHR-EHR-EHR_STATUS.generic.v1",
    "uid": {
      "_type": "HIER_OBJECT_ID",
      "value": "27e874f7-4371-4a35-8776-0c06f08cb196"
    },
    "is_modifiable": true,
    "is_queryable": true
  },
  "time_created": "2020-04-03T16:11:51.248370"
}
```

**Comment by nr23730** [comment]
@chevalleyc This is related to your changes at 97d9324c59d5cfb884eff7ef9d532720f175f241 in service/src/main/java/org/ehrbase/aql/sql/queryImpl/attribute/partyref/PartyResolver.java

Reverting the changes in that file makes the query work again. But that's for sure not how it's meant to be...

**Comment by Geisterfalle** [comment]
I just checked this. It looks like there is no issue regarding null but the partial paths for `e/ehr_status/subject/external_ref/id` still is not correct. Can you confirm @chevalleyc @nr23730 ?

**Comment by nr23730** [comment]
Yes, I can confirm this. 
I made a PR some time ago (#229) that had the branch of #228 as target that partially addressed this issue. But looks like #228 was merged into master before #229 was merged into the branch of #228. I can create another PR (with master as target this time!) if you like...

**Comment by Geisterfalle** [comment]
This would be fantastic, thx!
